app/services/email.py

```
@@ -0,0 +1,221 @@
+            email.sent_at = datetime.now(timezone.utc)
+        
+        # Update audit log
+        audit_log = []
+        if email.audit_log:
+            try:
+                audit_log = json.loads(email.audit_log)
+            except json.JSONDecodeError:
+                logger.error(
+                    "Corrupted audit_log for email %s, resetting to empty list",
+                    email_id,
+                )
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)